src/dfx/src/commands/replica.rs

```diff
@@ -178,8 +189,39 @@ pub fn exec(
     system.block_on(async move {
         let shutdown_controller = start_shutdown_controller(env)?;
         if emulator {
+            let effective_config = CachedConfig {
+                replica_rev: env!("DFX_ASSET_REPLICA_REV").into(),
+                config: CachedReplicaConfig::Emulator,
+            };
```

Comment:
There's some opportunity for DRY here.
```suggestion
            let effective_config = CachedConfig::emulator();
```